### ISSUE TITLE
Microk8s experiment build fail

### ISSUE THREAD

**Issue by valeriano-manassero** [polyaxon/polyaxon#457]
Tried to install latest Helm chart containing version 0.4.4 on **microk8s 1.14**  with this config:

```
rbac:
  enabled: false

ingress:
  enabled: false

serviceType: NodePort

postgresql:
  persistence:
    enabled: true
    size: 5Gi
```

Everything is ok but I cannot build:
```
2019-05-16 16:32:29 UTC -- Traceback (most recent call last):

2019-05-16 16:32:29 UTC -- File "dockerizer/build_cmd.py", line 37, in <module>

2019-05-16 16:32:29 UTC -- nocache=nocache)

2019-05-16 16:32:29 UTC -- File "/polyaxon/polyaxon/dockerizer/dockerizer/builders/native.py", line 151, in build

2019-05-16 16:32:29 UTC -- image_tag=image_tag)

2019-05-16 16:32:29 UTC -- File "/polyaxon/polyaxon/dockerizer/dockerizer/builders/native.py", line 33, in __init__

2019-05-16 16:32:29 UTC -- self.docker = APIClient(version='auto')

2019-05-16 16:32:29 UTC -- File "/usr/local/lib/python3.6/site-packages/docker/api/client.py", line 190, in __init__

2019-05-16 16:32:29 UTC -- self._version = self._retrieve_server_version()

2019-05-16 16:32:29 UTC -- File "/usr/local/lib/python3.6/site-packages/docker/api/client.py", line 215, in _retrieve_server_version

2019-05-16 16:32:29 UTC -- 'Error while fetching server API version: {0}'.format(e)

2019-05-16 16:32:29 UTC -- docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', ConnectionRefusedError(111, 'Connection refused'))
```

I see someone referring to internal registry not being reachable but my workarounds didn't work. Any suggestion?

**Comment by valeriano-manassero** [comment]
I finally figured out the issue:
builder tries to access Docker daemon trough mounted host `/var/run/docker.sock`.

This requires microk8s exposing the docker unix socket. A `microk8s.enable registry` should do the trick but the file is `/var/snap/microk8s/current/docker.sock`, so  a symlink is needed to have the right path.

But this is true in microk8s 1.13, according to https://github.com/ubuntu/microk8s/commit/7c3311a6d24a0eae482dc09e86471ead41aaa19e , microk8s switched to containerd so there is no more socket file.

A quick and dirty workaround is to install Docker directly on the host and enable unix socket adding `-H unix:///var/run/docker.sock` option.

@mouradmourafiq wdyt?

**Comment by mouradmourafiq** [comment]
you can also use `kaniko` as a backend for builds, [see](https://docs.polyaxon.com/integrations/kaniko/)

**Comment by valeriano-manassero** [comment]
> you can also use `kaniko` as a backend for builds, [see](https://docs.polyaxon.com/integrations/kaniko/)

Kaniko worked like a charm!

btw I needed to specify the build backend in experiment yaml file since setting default with `buildBackend: kaniko` in helm charts is ignored. I didn't investigate deeply to check if it's microk8s related (but it should not).

**Comment by mouradmourafiq** [comment]
> buildBackend: kaniko in helm charts is ignored.
Weird, will check that for next release.

**Comment by mouradmourafiq** [comment]
Ok default build backend had an issue, it fixed and it will be part of next release v0.5.
Closing this issue.
